lib/Arch/SPARC64_C.cpp

```diff
@@ -0,0 +1,452 @@
+  bool injected_sret = false;
+  auto err = BindReturnValues(func, injected_sret, fdecl.returns);
+  if (remill::IsError(err)) {
+    return err;
+  }
+
+  err = BindParameters(func, injected_sret, fdecl.params);
+  if (remill::IsError(err)) {
+    return err;
+  }
+
+  fdecl.return_stack_pointer_offset = 0;
+  fdecl.return_stack_pointer = arch->RegisterByName("o6");
+
+  fdecl.return_address.reg = arch->RegisterByName("o7");
+  fdecl.return_address.type = fdecl.return_address.reg->type;
+
+  return llvm::Error::success();
+}
+
+llvm::Error SPARC64_C::BindReturnValues(
+    llvm::Function &function, bool &injected_sret,
+    std::vector<anvill::ValueDecl> &ret_values) {
+
+  llvm::Type *ret_type = function.getReturnType();
+  injected_sret = false;
+
+  // If there is an sret parameter then it is a special case. For the X86_C ABI,
+  // the sret parameters are guarenteed the be in %eax. In this case, we can
+  // assume the actual return value of the function will be the sret struct
+  // pointer.
+  if (function.hasStructRetAttr()) {
+    ret_values.emplace_back();
+    auto &value_declaration = ret_values.back();
+
+    // Check both first and second parameter because llvm does that in
+    // llvm::Function::hasStructRetAttr()
+    if (function.hasParamAttribute(0, llvm::Attribute::StructRet)) {
+      value_declaration.type =
+          remill::NthArgument(&function, 0)->getType()->getPointerElementType();
+
+    } else if (function.hasParamAttribute(1, llvm::Attribute::StructRet)) {
+      value_declaration.type =
+          remill::NthArgument(&function, 1)->getType()->getPointerElementType();
+    }
+
+    value_declaration.type = llvm::PointerType::get(
+        value_declaration.type, 0);
+
+    if (!ret_type->isVoidTy()) {
+      return llvm::createStringError(
+          std::errc::invalid_argument,
+          "Function '%s' with sret-attributed parameter has non-void return type '%s'",
+          function.getName().str().c_str(),
+          remill::LLVMThingToString(ret_type).c_str());
+    }
+
+    value_declaration.reg = arch->RegisterByName("o0");
+    return llvm::Error::success();
+  }
+
+  switch (ret_type->getTypeID()) {
+    case llvm::Type::VoidTyID:
+      return llvm::Error::success();
+
+    case llvm::Type::IntegerTyID: {
+      const auto *int_ty = llvm::dyn_cast<llvm::IntegerType>(ret_type);
+      const auto int32_ty = llvm::Type::getInt32Ty(int_ty->getContext());
+      const auto bit_width = int_ty->getBitWidth();
+
+      if (bit_width <= 64) {
+        ret_values.emplace_back();
+        auto &value_declaration = ret_values.back();
+        value_declaration.reg = arch->RegisterByName("o0");
+        value_declaration.type = ret_type;
+        return llvm::Error::success();
+
+      // Split the integer across `o5:o0`. Experimentally, the largest
+      // returnable integer is 384 bits in size, any larger and LLVM crashes.
+      } else if (bit_width <= 384) {
```

Comment:
Create a constant for 384

Reply:
This isn't necessary imo; it's 6*64, i.e. the size of `o5:o4:o3:o2:o1:o0`.